client/src/templates/Challenges/components/scene/scene.tsx

```diff
@@ -131,9 +131,40 @@ export function Scene({
     }
   }
 
+  const buildTranscript = () => {
+    let transcript = '';
+    commands.forEach(command => {
+      console.log(command);
+      if (
+        command.character != null &&
+        command.dialogue != null &&
+        command.startTime != null
```

Comment:
```suggestion
        command.character &&
        command.dialogue &&
        command.startTime
```

I'm not sure why we're null-checking instead of truthy checking. If we *really* need to null check, we should at least use strict inequality:

```suggestion
        command.character !== null &&
        command.dialogue !== null &&
        command.startTime !== null
```